src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacFieldAndReferencedFieldMismatch/DacFieldAndReferencedFieldMismatchAnalyzer.cs

```diff
@@ -0,0 +1,220 @@
+{
+	public class DacFieldAndReferencedFieldMismatchAnalyzer : DacAggregatedAnalyzerBase
+	{
+		private readonly Dictionary<INamedTypeSymbol, DacSemanticModel?> _modelsCache = new(SymbolEqualityComparer.Default);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create(
+				Descriptors.PX1078_TypesOfDacFieldAndReferencedFieldMismatch,
+				Descriptors.PX1078_TypesOfDacFieldAndReferencedFieldHaveDifferentSize,
+				Descriptors.PX1078_DacFieldHasDataTypeAttributesIncompatibleWithReferencedFieldsCLRType);
+
+		public override void Analyze(SymbolAnalysisContext context, PXContext pxContext, DacSemanticModel dac)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			var propertiesToCheck = dac.DeclaredDacFieldPropertiesWithAcumaticaAttributes
+				.Select(p => (p, TryGetForeignReferenceAttribute(pxContext, p)))
+				.Where(p => p.Item2 is not null);
+
+			foreach (var (property, foreignReferenceAttribute) in propertiesToCheck)
+			{
+				context.CancellationToken.ThrowIfCancellationRequested();
+
+				CheckProperty(context, property, foreignReferenceAttribute!, dac, pxContext);
+			}
+		}
+
+		private static DacFieldAttributeInfo? TryGetForeignReferenceAttribute(PXContext pxContext, DacPropertyInfo property) =>
+			property.Attributes.FirstOrDefault(x => ContainsAttributeMatching(x, pxContext.AttributeTypes.PXSelectorAttribute.Type));
+
+		private void CheckProperty(SymbolAnalysisContext context, DacPropertyInfo property, DacFieldAttributeInfo foreignReferenceAttribute,
+			DacSemanticModel dac, PXContext pxContext)
+		{
+			var foreignField = ExtractForeignFieldSymbol(foreignReferenceAttribute, pxContext);
+
+			if (foreignField is null)
+				return;
+
+			var parentType = (foreignField.ContainingSymbol as INamedTypeSymbol)!;
+
+			var foreignDac = GetOrInferModel(parentType, pxContext, cancellation: context.CancellationToken);
+
+			if (foreignDac?.PropertiesByNames.TryGetValue(foreignField.Name, out var foreignProperty) != true)
+				return;
+
+			var foreignCLRType = foreignProperty!.PropertyType;
+
+			if (HaveDifferentUnderlyingType(property, foreignCLRType))
+				ReportTypeMismatch(context, pxContext, property, foreignProperty);
+			else
+			{
+				var localTypeAttributes = property.Attributes.Where(
+					x => ContainsAttributeMatching(x, pxContext.FieldAttributes.PXDBFieldAttribute)
+					|| ContainsAttributeMatchingOneOf(x, pxContext.FieldAttributes.UnboundDataTypeAttributes))
+					.ToList(property.Attributes.Length);
+
+				if (localTypeAttributes.Count == 0)
+					return;
+
+				var incompatibleAttribute = localTypeAttributes
+					.FirstOrDefault(attr => IsAttributeIncompatibleWithCLRType(attr, foreignCLRType, pxContext));
+
+				if (incompatibleAttribute is not null)
+					ReportTypeIncompatibility(context, pxContext, incompatibleAttribute, property, foreignProperty);
+
+				var foreignTypeAttribute = foreignProperty!.Attributes.FirstOrDefault(
```

Comment:
The local DAC field now may have multiple data type attributes. 
The foreign DAC field can also have multiple data type attributes but this isn't considered here.

Reply:
This remark was not fixed. For a local field you consider all data type attributes because they may actually have different specified sizes:
![image](https://github.com/user-attachments/assets/8b5a10c3-f414-4f8c-aeb1-7e68af114d2a)
And this is correct. But for foreign data type attributes you don't do this, keeping only the first foreign data type attribute.

In fact, I would first consolidate all field sizes from the attribute declared on a field, the `GetFieldSize` method seems like a good place to do it. If there are multiple sizes declared for a local field - return null and stop the analysis of the property. Do the same for a foreign field. Then there are just two sizes to compare.